src/Sylius/Bundle/CoreBundle/Theme/ChannelBasedThemeContext.php

```diff
@@ -22,26 +22,31 @@
 
 final class ChannelBasedThemeContext implements ThemeContextInterface
 {
+    private null|false|ThemeInterface $theme = false;
```

Comment:
A separate property for boolean would be better IMO eg. `$alreadyCalled`.

Reply:
I disagree. `false` means "no theme", `null` means not set, and the `ThemeInterface` means we have one.

All 3 values are logical.

Reply:
Hmmm I see the difference between the 3 of them but `no theme` also means nothing has been set yet, and `not set` means there is `no theme` so it's tricky at first glance – at least for me :smile:

Reply:
Defaulting it to false is the closest thing PHP has to an `is_initialized($this->prop);` type of check.  If it's false, the property hasn't been initialized yet and whatever code needs to seed it needs to run.  Once it's null or an object, it's initialized.

Reply:
I'm ok with the current implementation if anybody asks me 😂